inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -568,6 +568,75 @@ void ReversePermutationsPass::run() {
     }
 }
 
+void ConvolutionNCHWPass::run() {
+    int numOfadditionalPermuteLayers = 0;
+    auto quantized = InferenceEngine::getInjectedData<QuantizedLayerParams>(pLayers->front());
+
+    for (auto& l : *pLayers) {
+        IE_ASSERT(l != nullptr);
+        if (!LayerInfo(l).isConvolution()) {
+            continue;
+        }
+        auto inputs = l->insData.begin()->lock();
+        auto layerBeforeConv = CNNNetPrevLayer(l);
+        auto layerAfterConv = !getInputTo(l->outData.front()).empty() ? getInputTo(l->outData.front()).begin()->second : nullptr;
+
+        // TODO: add check for convolution with H != 1 and W != 1
+        // TODO: add check for convolution with H != 1 and W == 1
+        // required for Reshape1DOps transformation
+        if (layerBeforeConv->name.find("reshape_begin") != std::string::npos &&
```

Comment:
where does this reshape_begin come from? if same text is used in multiple passes make it a constant outside of pass (if possible)

Reply:
https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/reshape_1d_ops.cpp#L124 - I'm referring, for example, to this test: GNALayerTests/GNAConv1DTest.SplitToConcatWith2Inputs.

Reply:
This seems to be some ngraph specific code, Our plugin should not be dependant on something inserted by ngraph in their models, if this was name set was done by IE or GNA pass then it would be acceptable